server/controllers/roleController.js

```diff
@@ -1,77 +1,106 @@
 import db from '../models';
 
 const RoleController = {
-  createNewRole(req, res) {
-    let roleInfo = {};
-    if (req.body.id) {
-      roleInfo = { title: req.body.title, id: req.body.id };
-    } else { roleInfo = { title: req.body.title }; }
-    db.roles
-      .create(roleInfo)
+  createNewRole(request, response) {
+    let roleData = {};
+    roleData = { title: request.body.title };
+    if (!request.body.title) {
+      return response.status(400).send({ message: 'Title cannot be blank' });
+    }
+    db.roles.findOne({
+      where: { title: request.body.title }
+    })
+    .then((result) => {
+      if (result) {
+        return response.status(409).send({
+          success: false,
+          message: 'Role already exists'
+        });
+      }
+      return db.roles
+      .create(roleData)
       .then((role) => {
-        res.status(200).send({
+        response.status(200).send({
           message:
-          'The role was created successfully',
+          'The role was successfully created',
           role
         });
       })
       .catch((err) => {
-        res.status(400).send({ message: 'error', err });
+        response.status(400).send({ message:
+          'There was a error creating this role',
+          err });
       });
+    });
   },
 
-  updateRole(req, res) {
+  updateRole(request, response) {
     db.roles
-      .findById(req.params.id)
+      .findById(request.params.id)
       .then((role) => {
         if (!role) {
-          return res.status(404).send({ message: 'Role was not found' });
+          return response.status(404).send({ message:
+            'Cannot update a role that does not exist' });
+        }
+        const title = request.body.title;
+        if (!title) {
+          return response.status(404).send({ message:
+            'You need to write the Title you want to update' });
         }
         role.update({
-          title: req.body.title || role.title
+          title: request.body.title || role.title
         })
         .then((updatedRole) => {
-          res.status(200).send({ message: updatedRole });
+          response.status(200).send({
+            message: 'Role was successfully updated',
+            data: updatedRole });
         });
       });
   },
 
-  deleteRole(req, res) {
+  deleteRole(request, response) {
```

Comment:
What happens when I delete the admin role. Remember there is a foreign key constraint on a role to a user.  the only way this will work is if you're cascading the delete. This means the admin gets deleted, or all users and documents assigned to that role get wiped off. Just drawing this your attention.